src/main/java/net/openhft/chronicle/network/TcpEventHandler.java

```diff
@@ -197,6 +197,11 @@ public boolean action() throws InvalidEventHandlerException {
         if (tcpHandler == null)
             return false;
 
+        if (tcpHandler.hasTimedOut()) {
+            closeAndStartReconnector();
+            return false;
+        }
```

Comment:
We MUST close the TcpEventHandler prior to starting the reconnector, that's why we couldn't do it from HeartbeatCheckHandler, it has no reference to the TcpEventHandler